src/client/firefox/commands.js

```diff
@@ -317,6 +319,12 @@ async function fetchSources() {
   return sources.map(source => createSource(source, { supportsWasm }));
 }
 
+async function getDeviceDescription() {
+  const listTabsForm = await debuggerClient.listTabs();
```

Comment:
`listTabs` can actually be expensive to call, so it's best to avoid it if we can.

The `Target` has the root form cached as `target.root`, and I believe that should work for your needs here.

Reply:
Here's more detail: https://searchfox.org/mozilla-central/source/devtools/client/framework/target.js#282

It looks like it's not currently listed on the debugger's target type info, so you may need to add it there first.